File: CodeChef/Beginner/13LeadGame/leadgame.java
import java.util.Scanner;

class leadgame {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n, a, b, p1 = 0, p2 = 0, max = 0, winner = 0;
        n = sc.nextInt();

        for (int i = 0; i < n; i++) {
            a = sc.nextInt();
            b = sc.nextInt();

            p1 += a;
            p2 += b;

            if (p1 > p2 && (p1 - p2) > max) {
                max = p1 - p2;
                winner = 1;
            } else if (p2 > p1 && (p2 - p1) > max) {
                max = p2 - p1;
                winner = 2;
            }
        }
        sc.close();
        System.out.println(winner + " " + max);
    }
}

File: CodeChef/Beginner/13LeadGame/leadgame.cpp
#include <iostream>
using namespace std;
class leadgame {
public:
};
int main() {
  int n, a = 0, b = 0, a1, b1, max = 0, worl;
  cin >> n;
  for (int i = 0; i < n; i++) {
    cin >> a1;
    cin >> b1;

    a += a1;
    b += b1;

    if (a > b && (a - b) > max) {
      max = a - b;
      worl = 1;
    } else if (a < b && (b - a) > max) {
      max = b - a;
      worl = 2;
    }
  }
  cout << worl << " " << max << endl;
}